exercises/raindrops/instructions.md

```diff
@@ -0,0 +1,23 @@
+# Instructions
+
+Your task is to convert a number into its corresponding raindrop sounds.
+
+If a given number:
+
+- is divisible by 3, add "Pling" to the result.
+- is divisible by 5, add "Plang" to the result.
+- is divisible by 7, add "Plong" to the result.
+- **is not** divisible by 3, 5, or 7, the result should be the digits of the number.
```

Comment:
```suggestion
- **is not** divisible by 3, 5, or 7, the result should be the number as a string.
```

"the digits of the number" sounds like some (ordered) collection of the digits. But the result should definitely be a string.

Reply:
I think that we had decided to be vague on the "type" of the return for this, to allow a language to determine what they need to test for.

However, the results are a ordered collection of the digits when a string as the order must be most significant to the left and least significant digit to the right, and in that order.

Reply:
> "the digits of the number" sounds like some (ordered) collection of the digits. 

Yeah, I thought this when I was splitting it out, but figured it had been left vague for some reason for languages without strings. But then how do those languages do this exercise at all?

@kotp Do we have any concrete examples of where it's not a string?

Reply:
I do not have a concrete example, but remember talking about how even "Hello World" returning a string in some languages is not "simple".  And so leaving the answer to be a collection is well enough.  It has been some years.

Reply:
"How do they do this exercise at all?"  They might not.  However, having a collection of some kind can store those numeric values, and may also represent the strings as they see fit.  It is the relationship of the numbers being replaced by some value given some criteria that is important for the exercise, not really the strings values, I mean as far as learning value goes, and that does not diminish the "fun" value in my mind (think ciphergraphic kid games).

Reply:
My sense is that as this exercise is all about strings, we could safely get away with putting "string" in the description for this bit. I think it's much clearer if we can get away with it.

So I think let's fine a concrete counterexample, or go for Glenn's change. And if there's like only one counterexample, maybe that track should just make an exception?

Reply:
Right, just like all the exercises, the tracks are "inspired" by the problem-specifications  data, not ruled by it.

Reply:
If this exercise were a concept exercise, then I would consider that potentially it is all about strings.  In Ruby I usually mentor this as "all about construction and flow control" rather than focusing on the fact that they are strings.  This is what we concentrate on as an Interview question for the simpler version, as well as expansion, when I use this as an interview question.

Reply:
> So I think let's fine a concrete counterexample, or go for Glenn's change. And if there's like only one counterexample, maybe that track should just make an exception?

Not against Glenn's change, just the reason for it is not disqualifying for what is there.  (It is not a contradictory thing.)

Reply:
Yeah. OK - I'm going to go with it :) Thanks!